src/rt/sched/cown.h

```diff
@@ -528,26 +530,38 @@ namespace verona::rt
       const auto high_priority = behaviour_requires_high_priority(body);
       for (; body->index < body->count; body->index++)
       {
-        MultiMessage* m = MultiMessage::make_message(alloc, body, epoch);
+        auto* m = MultiMessage::make_message(alloc, body, epoch);
         auto* next = body->cowns[body->index];
         Systematic::cout() << "MultiMessage " << m << ": fast requesting "
                            << next << ", index " << body->index << std::endl;
 
-        {
-          // Hold epoch in case priority needs to be raised after message is
-          // placed in queue. TODO: only hold if `high_priority` is true
-          Epoch e(alloc);
+        if (body->index > 0)
+          body->cowns[body->index - 1]->blocker = next;
```

Comment:
I think you need to check if 
```
body->cowns[body->index - 1]
```
has a `Priority::High`, and use that for `high_priority`.